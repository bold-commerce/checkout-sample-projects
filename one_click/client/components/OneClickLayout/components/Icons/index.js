export { default as TimesCircle } from './TimesCirlce'
export { default as ArrowRight } from './ArrowRight'
export { default as CaretRight } from './CaretRight'