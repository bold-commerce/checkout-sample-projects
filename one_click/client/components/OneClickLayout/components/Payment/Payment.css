.FieldSet--PaymentMethod .FieldSet__Content iframe {
  width: 100%;
  height: 100%;
  border: none;
}

.FieldSet--PaymentMethod .FieldSet__Content iframe .ToggleField__Input:checked {
  box-shadow: 0 0 0 5px #0099cc inset;
}

section.FieldSet.FieldSet--PaymentMethod {
  margin: 0;
}

section.FieldSet.FieldSet--PaymentMethod .FieldSet__Content {
  margin-bottom: 16px;
}
