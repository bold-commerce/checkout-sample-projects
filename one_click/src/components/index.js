export * from './Address';
export * from './BackButton';
export * from './Billing';
export * from './BillingAddress';
export * from './Card';
export * from './CheckoutButton';
export * from './CheckoutForm';
export * from './Confirmation';
export * from './CustomerInfo';
export * from './Discount';
export * from './EmptyState';
export * from './Header';
export * from './Icons';
export * from './Index';
export * from './InventoryItem';
export * from './LineItems';
export * from './LoadingState';
export * from './Payment';
export * from '../pages/Processing';
export * from './Product';
export * from './RadioItem';
export * from './Shipping';
export * from './ShippingLines';
export * from './Summary';
