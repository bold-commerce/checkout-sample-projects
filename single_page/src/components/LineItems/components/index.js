export { default as LineItem } from './LineItem';
export { default as Product } from './Product';
