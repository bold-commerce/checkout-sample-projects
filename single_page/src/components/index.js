export * from './Address';
export * from './BillingAddress';
export * from './CheckoutButton';
export * from './CheckoutSection';
export * from './Customer';
export * from './DiscountForm';
export * from './EmptyState';
export * from './Header';
export * from './LineItems';
export * from './LoadingState';
export * from './OrderErrors';
export * from './OrderSummary';
export * from './PaymentMethod';
export * from './ShippingAddress';
export * from './ShippingLines';
