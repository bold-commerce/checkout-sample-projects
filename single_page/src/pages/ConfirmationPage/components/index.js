export * from './ConfirmationList';
export * from './ConfirmationListItem';
