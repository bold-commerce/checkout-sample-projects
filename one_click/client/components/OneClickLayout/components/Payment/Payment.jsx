import React from 'react';
import PropTypes from 'prop-types';
import EmptyState from '../EmptyState/EmptyState';
import { BackButton } from '../BackButton';
import { usePaymentMethod } from '@boldcommerce/checkout-react-components';
import PaymentIframe from './PaymentIframe';
import './Payment.css';

export const PaymentMethod = ({ showPaymentMethod }) => {
  if (showPaymentMethod) {
    return (
      <div className="Payment">
        <BackButton />
        <section className="FieldSet FieldSet--PaymentMethod">
          <div className="FieldSet__Header">
            <div className="FieldSet__Heading">Payment methods</div>
          </div>
          <div className="FieldSet__Content">
            <EmptyState title="To view payment options, complete filling in your address" />
            <PaymentIframe key="paymentIframe" hide />
          </div>
        </section>
      </div>
    );
  }

  return (
    <section className="FieldSet FieldSet--PaymentMethod">
      <div className="FieldSet__Header">
        <div className="FieldSet__Heading">Payment methods</div>
      </div>
      <div className="FieldSet__Content">
        <PaymentIframe key="paymentIframe" />
      </div>
    </section>
  );
};

PaymentMethod.propTypes = {
  showPaymentMethod: PropTypes.bool,
};

const MemoizedPaymentMethod = React.memo(PaymentMethod);

const PaymentMethodContainer = () => {
  const { showPaymentMethod } = usePaymentMethod();

  return <MemoizedPaymentMethod showPaymentMethod={showPaymentMethod} />;
};

export default PaymentMethodContainer;
