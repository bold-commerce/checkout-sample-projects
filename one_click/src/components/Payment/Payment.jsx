import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import EmptyState from '../EmptyState/EmptyState';
import PaymentIframe from './PaymentIframe';
import { Discount } from '../Discount';
import { useLocation } from 'react-router-dom';
import { BillingAddress } from '../BillingAddress';
import { useBillingAddress, useShippingLines } from '@boldcommerce/checkout-react-components';
import './Payment.css';
import { useTranslation } from 'react-i18next';

const PaymentMethod = ({ billingAddress, shippingLines }) => {
  const location = useLocation();
  const hidePaymentComponent = !(location.pathname === "/");
  const disabled = Array.isArray(billingAddress) && shippingLines.length === 0;
  const { t } = useTranslation();

  if (disabled) {
    return (
      <div className='Payment'>
        <section className={classnames(
          {'hidden': hidePaymentComponent ,
          'FieldSet FieldSet--PaymentMethod': true
        })}>
          <div className="FieldSet__Content">
            <EmptyState title={t('payment.empty')} />
            <PaymentIframe key="paymentIframe" hide />
          </div>
        </section>
      </div>
    );
  }

  return (
    <>
      <div className='Payment'>
        <section className={classnames(
          {'hidden': hidePaymentComponent,
          'FieldSet FieldSet--PaymentMethod': true
        })}>
          <div className="FieldSet__Content">
            <PaymentIframe key="paymentIframe" />
          </div>
          <div className="FieldSet__Content">
            <Discount />
          </div>
          <div className="FieldSet__Content">
            <BillingAddress />
          </div>
        </section>
      </div>
    </>
  );
};

PaymentMethod.propTypes = {
  showPaymentMethod: PropTypes.bool,
};

const MemoizedPaymentMethod = React.memo(PaymentMethod);

const PaymentMethodContainer = () => {
  const { data: billingAddress } = useBillingAddress();
  const { data } = useShippingLines(); 
  const { shippingLines } = data;

  return <MemoizedPaymentMethod shippingLines={shippingLines} billingAddress={billingAddress} />;
};

export default PaymentMethodContainer;
