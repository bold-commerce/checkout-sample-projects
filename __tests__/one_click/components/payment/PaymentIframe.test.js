import React from 'react';
import { render } from '@testing-library/react';
import { PaymentIframe } from '../../../../one_click/src/components/Payment'
import { MemoryRouter } from "react-router";
import {
    exampleUseBillingSameAsShipping as MOCKbillingSameAsShipping,
    exampleUsePaymentMethod as MOCKpaymentMethod,
    exampleUsePaymentIframe as MOCKpaymentIframe,
    exampleUseCountryInfo as MOCKcountryInfo
} from '../../../utils/hookHelpers';

jest.mock('@boldcommerce/checkout-react-components', () => ({
    ...jest.requireActual('@boldcommerce/checkout-react-components'),
    useBillingSameAsShipping: () => MOCKbillingSameAsShipping,
    usePaymentMethod: () => MOCKpaymentMethod,
    usePaymentIframe: () => MOCKpaymentIframe,
    useCountryInfo: () => MOCKcountryInfo,
}))

describe('PaymentIframe', () => {
    test('renders PaymentIframe component', () => {
        const { asFragment } = render(
            <PaymentIframe />,
            {wrapper: MemoryRouter}
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
