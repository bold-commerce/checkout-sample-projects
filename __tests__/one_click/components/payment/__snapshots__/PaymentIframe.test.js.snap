// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentIframe renders PaymentIframe component 1`] = `
<DocumentFragment>
  <iframe
    data-bold-pigi-iframe="true"
    src="test.url"
    title="payments"
  />
</DocumentFragment>
`;
